routemaster/mixins/redis.rb

```diff
@@ -12,13 +11,10 @@ def self.included(by)
       end
 
       def _redis
-        @@_redis ||= ::Redis::Namespace.new(
-          ENV.fetch('ROUTEMASTER_REDIS_PREFIX'),
-          redis: ::Redis.new(url: ENV['ROUTEMASTER_REDIS_URL']))
-      end
-
-      def _redis_prefix
-        @@_redis_prefix ||= ENV.fetch('ROUTEMASTER_REDIS_PREFIX')
+        @@_redis ||=
+        ConnectionPool.wrap(size: Routemaster.config[:redis_pool_size], timeout: 2) do
```

Comment:
I considered this; but I don't think it'd be that useful, because:

- most of the Redis usage is Lua scripts (there rarely is a sequence of calls that'd benefit from being made from the same connection.
- other places use `multi`, which is implicitly equivalent (it holds a lock on the connection). 
- the cost of the pool locking is minimal compared to the Redis roundtrip.

Let me know if that convinces you.